### PR TITLE
feat: color knob

### PR DESCRIPTION
**Pull request by mulieriq**
### ColorKnob
- Added Color Knob
- Added Test

### List of issues which are fixed by the PR
*You must list at least one issue.*

### Screenshots
*If applicable, add screenshots to help explain the changes.*

### Checklist

- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on [Discord].

[CLA]: https://docs.google.com/forms/d/e/1FAIpQLScuRfjUzENsLsmQgqZlGLxMKbFi7zuXoPARyXytoyQrq7ntUw/viewform
[Discord]: https://discord.com/invite/zT4AMStAJA

### CODE REVIEW

File: packages/widgetbook/lib/src/knobs/slider_knob.dart [hunk 1]
```diff
@@ -5,17 +5,13 @@ import 'package:widgetbook/src/knobs/nullable_checkbox.dart';
...
+    required super.value,
     this.max = 1,
     this.min = 0,
     this.divisions,
-  }) : super(
-          label: label,
-          description: description,
-          value: value,
-        );
+  });
```
Comment (jenshor) [inline packages/widgetbook/lib/src/knobs/slider_knob.dart:14]: This should not be part of this PR, but we can leave it in as it improves the code 👍🏻

File: packages/widgetbook/lib/src/knobs/knobs_builder.dart [hunk 2]
```diff
@@ -10,6 +11,15 @@ abstract class KnobsBuilder {
     bool initialValue = false,
   });
 
+// Creates a textfield that can be typed in and optionally hold a
+  // color value
+
```
Comment (jenshor) [inline packages/widgetbook/lib/src/knobs/knobs_builder.dart:16]: Comment has improper formatting.

File: packages/widgetbook/lib/src/knobs/color_knob.dart [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+const _kFallbackColor = 0xFFFF0000;
+
+extension on String {
+  Color asColor() {
+    if (length != 7) {
+      return const Color(_kFallbackColor);
+    }
+    return Color(int.tryParse(replaceFirst('#', '0xFF')) ?? _kFallbackColor);
+  }
+}
```
Comment (jenshor) [inline packages/widgetbook/lib/src/knobs/color_knob.dart:14]: Ok for now, but I think having a fallback color is probably a bit weird from a UX perspective.

**Review by jenshor — CHANGES_REQUESTED** [review]
Besides the one comment it looks good to me 👍🏻

File: packages/widgetbook/lib/src/knobs/knobs_builder.dart [hunk 4]
```diff
@@ -10,6 +11,15 @@ abstract class KnobsBuilder {
     bool initialValue = false,
   });
 
+/// Creates a textfield that can be typed in and optionally hold a
+// color value
+
```
Comment (jenshor) [inline packages/widgetbook/lib/src/knobs/knobs_builder.dart:16]: This should not have an empty space between the comment and the code. This should be a docs comment which starts with `///` instead of `//` and the indentation should match the indentation of the code.

**Review by jenshor — CHANGES_REQUESTED** [review]

File: packages/widgetbook/lib/src/knobs/knobs_builder.dart (hunk 4 above)
Comment (mulieriq) [reply packages/widgetbook/lib/src/knobs/knobs_builder.dart]: Got it

File: packages/widgetbook/test/src/widgetbook_test.dart [hunk 5]
```diff
@@ -40,11 +40,19 @@ void main() {
...
-                () => Widgetbook(
-                  addons: const [],
+                () => Widgetbook<ThemeData>(
+                  addons: [
+                    WidgetbookAddOn(
+                      name: 'A',
+                      builder: (context) => const SizedBox(),
+                    ),
+                  ],
+                  ],
```
Comment (jenshor) [inline packages/widgetbook/test/src/widgetbook_test.dart:50]: The test file does not compile in general, but this syntax has one `]` too much.

File: packages/widgetbook/test/src/widgetbook_test.dart [hunk 6]
```diff
@@ -40,11 +40,19 @@ void main() {
...
                   categories: const [],
                   appInfo: appInfo,
-                  themes: themes,
+                  
+                  //addons:const []
+                //  themes: themes,
```
Comment (jenshor) [inline packages/widgetbook/test/src/widgetbook_test.dart:55]: This should also not be part of the PR.

**Review by jenshor — CHANGES_REQUESTED** [review]

File: packages/widgetbook/test/src/widgetbook_test.dart (hunk 6 above)
Comment (mulieriq) [reply packages/widgetbook/test/src/widgetbook_test.dart:55]: My bad, I had started working, on the other issues

**Review by jenshor — APPROVED** [review]
LGTM 🎉
